### PR TITLE
Add initial spdlog functionlity

### PR DESCRIPTION
**Pull request by a4z**
* Setup spdlog
* Adopt config for console color
* Map loglevels
* Add some loggin macros

spdlog writes into a file and to stderr.
Filename can already be configured.
Loglevels can be used.

It is now possible to write out messages via spdlog.

Attention: since spdlog uses the same config values like the
existin log for the log file, and the existing log is not
deactivated, there might be some colision in the messages.
###  :notebook_with_decorative_cover: Description

This is a draft, mirroring the first point of the the SOW
This PR acts as an early preview for having a chance for early feedback and to create a base of discussion of implementation details

###  :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [ ] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -508,6 +508,9 @@ find_package(yaml-cpp 0.6.2 REQUIRED HINTS ${YAML_CPP_ROOT})
 get_filename_component(yaml_cpp_dir "${YAML_CPP_INCLUDE_DIR}" ABSOLUTE)
 get_filename_component(yaml_cpp_dir "${yaml_cpp_dir}" DIRECTORY)
 
+
+find_package(spdlog 1.8.1 REQUIRED)
```
Comment (dominiklohmann) [inline CMakeLists.txt:512]:
```suggestion
find_package(spdlog 1.5 REQUIRED)
```

On Debian Buster, the newest available backport for `spdlog` is 1.5.0: https://packages.debian.org/buster-backports/libspdlog-dev 

Supporting Debian Buster is a necessary requirement for our deployments at the moment.

**Comment by lava** [comment]
Hm, the CI is currently failing because spdlog isn't installed on the runners. You could adjust the job definitions to install the dev packages before building, but afaik Github will not run modified job descriptions from "untrusted" PRs.

@dominiklohmann , do you have an idea how to work around this?

**Comment by dominiklohmann** [comment]
> @dominiklohmann , do you have an idea how to work around this?

We create a PR that does this and change this PR's base to that PR, which we essentially use as a merge base for all spdlog-related PRs.

**Comment by dominiklohmann** [comment]
I have changed the merge base of this PR to `epic/spdlog` (#1223). This also allows us to use smaller, more reviewable PRs for your work @a4z, because we can merge them independently and will only merge the `epic/spdlog` branch in the very end into `master`.

**Comment by mavam** [comment]
Hey @a4z, mind giving our [contributing guidelines](https://github.com/tenzir/.github/blob/master/contributing.md#commit-messages) a read? We have a very structured approach and it is important to us that contributions that are not one-offs fit in there. For example, we try to strive for minimal vertical whitespace and capitalize git commit message summaries.

**Comment by a4z** [comment]
sorry @mavam , I will  ( maybe should already now) y close this PR, since it it not even a PR (draft) and I use this too much as a backup, so things land here that are not ment to be reviewed 

at the end, commits will of course be stashed

**Comment by mavam** [comment]
No worries. Just making sure that all relevant information regarding the process is on your radar from the get-go.

File: CMakeLists.txt (hunk 1 above)
Comment (a4z) [reply CMakeLists.txt]: resolved in the latest changes

File: libvast/vast/logger.hpp [hunk 2]
```diff
@@ -13,13 +13,51 @@
...
 #pragma once
 
-#include "vast/config.hpp"
 #include "vast/detail/discard.hpp"
-#include "vast/detail/pp.hpp"
-#include "vast/detail/type_traits.hpp"
 
-#include <caf/detail/pretty_type_name.hpp>
-#include <caf/logger.hpp>
+// from chat .. TODO, verify
```
Comment (a4z) [inline libvast/vast/logger.hpp:18]: Schöner machen

**Comment by lava** [comment]
Closing this in favor of https://github.com/tenzir/vast/pull/1293

**Pull request closed without merging**